securitas/controller/password.py

```diff
@@ -1,44 +1,75 @@
-from flask import flash, render_template, redirect, url_for
+from flask import flash, render_template, redirect, url_for, session
 import python_freeipa
 
 from securitas import app
-from securitas.security.ipa import untouched_ipa_client
+from securitas.security.ipa import untouched_ipa_client, maybe_ipa_session
+from securitas.utility import with_ipa
 from securitas.form.password_reset import PasswordResetForm
 
 
+def _validate_change_pw_form(form, ipa=None):
+    if ipa is None:
+        ipa = untouched_ipa_client(app)
+
+    username = form.username.data
+    current_password = form.current_password.data
+    password = form.password.data
+
+    res = None
+    try:
+        res = ipa.change_password(username, password, current_password)
+    except python_freeipa.exceptions.PWChangeInvalidPassword:
+        form.current_password.errors.append(
+            "The old password or username is not correct"
+        )
+    except python_freeipa.exceptions.PWChangePolicyError as e:
+        form.password.errors.append(e.policy_error)
+    except python_freeipa.exceptions.FreeIPAError as e:
+        # If we made it here, we hit something weird not caught above. We didn't
+        # bomb out, but we don't have IPA creds, either.
+        app.logger.error(
+            f'An unhandled error {e.__class__.__name__} happened while reseting '
+            f'the password for user {username}: {e.message}'
+        )
+        form.errors['non_field_errors'] = ['Could not change password.']
+
+    if res and res.ok:
+        flash('Your password has been changed', 'success')
+        app.logger.info(f'Password for {username} was changed')
```

Comment:
I think it would be useful to also log something when the call was unsuccessful.

Reply:
is the logging we arfe doing in _validate_change_pw_form() not enough here? it logs if we hit some errors.

Not sure how to avoid logging all validation errors here, as these will show up here too, right?

Reply:
Yeah you're right, it's fine.